aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1304,7 +1304,9 @@ TORCH_IMPL_FUNC(mean_out)
   // (mean_kernel_impl()) is unvectorized and leads to very poor performance
   // for production workloads. Once that's fixed, the following code can be used
   // in lieu of the sum + divide implementation below.
-  if (self.device().is_cpu()) {
+  // Note: there has an accuracy loss for half and bfloat16 which has one more type
+  // cast compared with mean_stub path.
+  if (self.device().is_cpu() && dtype != kHalf && dtype != kBFloat16) {
```

Comment:
For ComplexHalf and ComplexBfloat16, both cuda and CPU seems doesn't use accumulate dtype for mean_stub, it may be better to change it for higher precision.